pallets/democracy-precompiles/src/lib.rs

```diff
@@ -0,0 +1,277 @@
+// Copyright 2019-2021 PureStake Inc.
+// This file is part of Moonbeam.
+
+// Moonbeam is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Moonbeam is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Moonbeam.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Precompile to interact with pallet democracy through an evm precompile.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use evm::{executor::PrecompileOutput, Context, ExitError, ExitSucceed};
+use frame_support::dispatch::{Dispatchable, GetDispatchInfo, PostDispatchInfo};
+use frame_support::traits::{Currency, Get};
+use pallet_evm::AddressMapping;
+use pallet_evm::GasWeightMapping;
+use pallet_evm::Precompile;
+use sp_core::{H160, U256};
+use sp_std::convert::{TryFrom, TryInto};
+use sp_std::fmt::Debug;
+use sp_std::marker::PhantomData;
+use sp_std::vec::Vec;
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+type BalanceOf<Runtime> = <<Runtime as pallet_democracy::Config>::Currency as Currency<
+	<Runtime as frame_system::Config>::AccountId,
+>>::Balance;
+
+/// A precompile to wrap the functionality from pallet democracy.
+///
+/// Grants evm-based DAOs the right to vote making them first-class citizens.
+///
+/// EXAMPLE USECASE:
+/// A political party that citizens delegate their vote to, and the party votes on their behalf.
+pub struct DemocracyWrapper<Runtime>(PhantomData<Runtime>);
+
+impl<Runtime> Precompile for DemocracyWrapper<Runtime>
+where
+	Runtime: pallet_democracy::Config + pallet_evm::Config,
+	Runtime::AccountId: From<H160>,
+	BalanceOf<Runtime>: TryFrom<U256> + Debug,
+	Runtime::Call: Dispatchable<PostInfo = PostDispatchInfo> + GetDispatchInfo,
+	<Runtime::Call as Dispatchable>::Origin: From<Option<Runtime::AccountId>>,
+	Runtime::Call: From<pallet_democracy::Call<Runtime>>,
+{
+	fn execute(
+		input: &[u8], //Reminder this is big-endian
+		target_gas: Option<u64>,
+		context: &Context,
+	) -> Result<PrecompileOutput, ExitError> {
+		log::trace!(target: "democracy-precompile", "In democracy wrapper");
+
+		// Basic sanity checking for length
+		// https://solidity-by-example.org/primitives/
+		const SELECTOR_SIZE_BYTES: usize = 4;
+
+		//TODO test for this
+		if input.len() < 4 {
```

Comment:
is `4` supposed to use the constant defined above?

Reply:
```suggestion
		if input.len() < SELECTOR_SIZE_BYTES {
```